### ISSUE TITLE
Passing "id" parameter on create service method allows for unrestricted "get" access to feathers-objection service

### ISSUE THREAD

**Issue by Leon-Sam** [feathersjs-ecosystem/feathers-objection#88]
Hi guys,

I'm not sure if this is a bug or an intended feature. At face value it seems like a security issue.

**My setup:** 
Feathers typescript app generated via Feathers CLI generator (V4.5.2)
Feathers-objection database adapter (V5.2.0)
Node 12.16.1
Postgres 12
Service endpoints created via Feathers CLI generator

**The problem:**
Using a REST client, I POST data to the service endpoint to create a row in the database. Everything works as intended.  

However, if I POST data with an "id" parameter. It will create a new entry in the database as intended, but it will return the data of the user provided "id". (I define the primary key "id" is in the $beforeInsert hook of the Objection Model.)

**Temporary workaround:**
Sanitize input for "id" parameter before it reaches the database adapter

**Where the potential bug was introduced:**
I dug through the git history, and it looks like the logic was introduced between v0.4 and v0.5.  Line 220 to be exact.
[Here is a link](https://github.com/feathersjs-ecosystem/feathers-objection/commit/c70e17c3911b39ece733610c58811d5208c7a29e#diff-1fdf421c05c1140f6d71444ea2b27638R219-R222) to the specific commit 

**Proposed fix:**
Remove the logic that defines the "id" parameter for the _get method via the "data" variable, and only allow the value in the "row" variable to set the "id"

**Risk levels:** 

- If your primary keys are generated via UUIDs, then it's pretty hard for end users to guess the keys at face value. But if your PK are exposed via your app. Then bad actors can bypass any security logic in the "find" or "get" service methods by just using the "create" service method.
- If the primary keys are sequentially incremented via the database, then a malicious actor can repeatably POST the service endpoint with incrementing id values, allowing them to scrape private data.
- If you generate your primary key "id" parameter  in a "before" feathers hook, then it would just overwrite the user provided "id" parameter, and there is no security issue.

**My question:**

What was the logic in following line? (This is line 220 from the previous link)
`      const id = typeof data[this.id] !== 'undefined' ? data[this.id] : row[this.id]
`

At face value the code above basically transforms a "create" service request into a leaky "get" service request.

**Proposed fix:**
Change line 220 (from the link) or line 485/488 (of v5.3.0)
`
 const id = row[this.id]
`
The code above  only returns the data from new entry created by the database.

Can someone double check my logic?

**Comment by dekelev** [comment]
Hi @Leon-Sam , Thanks for the detailed info. that logic was applied since not all supported DBs support the [Knex `returning` method](http://knexjs.org/#Builder-returning).

Personally, I'm always adding validations on the input data, when it comes to an external REST call. that also prevents a client from sending the `id` field, which is a bad-practice in general.

When I do need to set the `id` field before the INSERT query happens, I use a before hook in the service hooks file.

I use `$beforeInsert` model hook mainly for setting the updatedAt, createdAt and setting a field with value from another field.

**Comment by daffl** [comment]
Could we then just take out the `data[id]` parts in https://github.com/feathersjs-ecosystem/feathers-objection/blob/master/src/index.js#L479 ?

**Comment by Leon-Sam** [comment]
I totally agree with the best practice of sanitizing the inputs. I just noticed the bug while was building out the core service methods and before I added any sanitization logic. 

I did a bit of digging in Objection's API, It looks like they have[ InsertandFetch](https://vincit.github.io/objection.js/api/query-builder/mutate-methods.html#insertandfetch) variants for their method calls. I did a quick hail mary and updated the method calls, and removed the logic that both daffl and I mentioned. It works great on my end with a Postgres instance. 

But it's failing the test suite on " Composite PK Queries". (Line 537 on ./tests/index.test.js) It's probably the exact logic that  @dekelev  was mentioning with "returning" method.

[Here's current diff of the commit](https://github.com/Leon-Sam/feathers-objection/commit/6e50270588112a2e3883b1a8e140ed0d158ccdd3) that I'm working with. It's a bit late right now, I'm going to call it a night and figure this out once things at work calm down for me. Hopefully in a few days.
